rts/idris_rts.c

```diff
@@ -594,6 +594,16 @@ VAL idris_strIndex(VM* vm, VAL str, VAL i) {
     return MKINT((i_int)idx);
 }
 
+VAL idris_substr(VM* vm, VAL offset, VAL length, VAL str) {
+    char *start = idris_utf8_advance(GETSTR(str), GETINT(offset));
+    char *end = idris_utf8_advance(start, GETINT(length));
+    Closure* newstr = allocate(sizeof(Closure) + (end - start) +1, 0);
+    SETTY(newstr, STRING);
+    memcpy(newstr -> info.str, start, end - start);
+    end[1] = '\0';
```

Comment:
Isn’t end here a pointer into the original string, and thus this mutates the original string?